crawler/priority_samples_process.py

```diff
@@ -0,0 +1,267 @@
+        if dart_success:
+            # use stored identifiers to update priority_samples table to processed true
+            sample_ids = list(map(extract_mongo_id, samples))
+            update_unprocessed_priority_samples_to_processed(db, sample_ids)
+
+
+def query_any_unprocessed_samples(db: Database) -> List[Any]:
+    priority_samples_collection = get_mongo_collection(db, COLLECTION_PRIORITY_SAMPLES)
+
+    # Query:
+    # from the priority samples collection
+    # get all unprocessed samples, we want to update also samples that have changed their importance
+    # join on the samples collection using sample_id in priority_samples collection to _id in the samples collection
+    # flatten object so sample fields are at the same level as the priority samples fields, removing
+    # the nested sample object return a list of the samples
+    # e.g db.priority_samples.aggregate([{"$match":{"$and": [
+    # {"processed": false} ]}}, {"$lookup": {"from": "samples", "let": {"sample_id": "$_id"},"pipeline":
+    # [{"$match": {"$expr": {"$and":[{"$eq": ["$sample_id", "$$sample_id"]}]}}}], "as": "from_samples"}}])
+    IMPORTANT_UNPROCESSED_SAMPLES_MONGO_QUERY: Final[List[object]] = [
+        # first get only unprocessed priority samples
+        {
+            "$match": {
+                "$and": [
+                    {FIELD_PROCESSED: False},
+                ]
+            }
+        },
+        {
+            "$lookup": {
+                "from": "samples",
+                "localField": "sample_id",
+                "foreignField": FIELD_MONGODB_ID,
+                "as": "related_samples",
+            }
+        },
+        # replace the document with a merge of the original and the first element of the array created from the lookup
+        # above - this should always be 1 element
+        {"$replaceRoot": {"newRoot": {"$mergeObjects": [{"$arrayElemAt": ["$related_samples", 0]}, "$$ROOT"]}}},
+        # result = { mongo_sample_id=1234, must_sequence=true; related_samples: { uuid: 132 } }
+        # remove the lookup document
+        {"$project": {"related_samples": 0}}
+        # result = { mongo_sample_id=1234, must_sequence=true, uuid: 132 }
+    ]
+
+    value = priority_samples_collection.aggregate(IMPORTANT_UNPROCESSED_SAMPLES_MONGO_QUERY)
+    return list(value)
+
+
+def merge_priority_samples_into_docs_to_insert(priority_samples: List[Any], docs_to_insert: list) -> None:
+    """
+    Updates the sample records with must_sequence and preferentially_sequence values
+
+    for each successful add sample, merge into docs_to_insert_mlwh
+    with must_sequence and preferentially_sequence values
+
+    Arguments:
+        priority_samples  - priority samples to update docs_to_insert with
+        docs_to_insert {List[ModifiedRow]} -- the sample records to update
+    """
+
+    def extract_root_sample_id(sample: SampleDoc) -> SampleDocValue:
+        return sample[FIELD_ROOT_SAMPLE_ID]
```

Comment:
We have to change this to merge with sample_ids instead of root sample ids

Reply:
Should now be outdated